src/CKEditor.tsx

```diff
@@ -0,0 +1,223 @@
+import registerEditorEventHandler from './registerEditorEventHandler';
+import useCKEditor from './useCKEditor';
+import { camelToKebab } from './utils';
+
+import type {
+	CKEditorEditorEventName,
+	CKEditorEventDispatcher,
+	CKEditorEventHandlerName,
+	CKEditorEventHandlersProps,
+	CKEditorProps,
+	CKEditorType
+} from './types';
+
+const { useEffect, useRef, useState } = React;
+
+/**
+ * `CKEditor` component is a convenient wrapper around low-level hooks.
+ * It's useful for simpler use cases. For advanced usage see `useCKEditor` hook.
+ */
+function CKEditor( {
+	config,
+	debug,
+	editorUrl,
+	initData,
+	name,
+	readOnly,
+	style,
+	type,
+
+	/**
+	 * `handlers` object must contain event handlers props only!
+	 */
+	...handlers
+}: CKEditorProps ): JSX.Element {
+	/**
+	 * Uses `useState` instead of `useRef` to force re-render.
+	 */
+	const [ element, setElement ] = useState<HTMLDivElement | null>( null );
+
+	/**
+	 * Ensures referential equality of event handlers.
+	 */
+	const refs = useRef<CKEditorEventHandlersProps>( handlers );
+
+	const dispatchEvent: CKEditorEventDispatcher = ( { type, payload } ) => {
+		const handlerName = eventNameToHandlerName( type );
+		const handler = refs.current[ handlerName ];
+
+		if ( handler ) {
+			handler( payload );
+		}
+	};
+
+	/**
+	 * `readOnly` prop takes precedence over `config.readOnly`.
+	 */
+	if ( config && typeof readOnly === 'boolean' ) {
+		config.readOnly = readOnly;
+	}
+
+	const { editor, status } = useCKEditor( {
+		config,
+		dispatchEvent,
+		debug,
+		editorUrl,
+		element,
+
+		/**
+		 * Subscribe only to those events for which handler was supplied.
+		 */
+		subscribeTo: Object.keys( handlers ).map( key => {
+			const evtName = key.substr( 2, 1 ).toLowerCase() + key.substr( 3 );
+			return evtName as CKEditorEditorEventName;
+		} ),
+		type
+	} );
+
+	/**
+	 * Sets and updates styles.
+	 */
+	useEffect( () => {
+		const canSetStyles =
+			type !== 'inline' &&
+			editor &&
+			( status === 'loaded' || status === 'ready' );
+
+		if ( style && canSetStyles ) {
+			editor.container.setStyles( style );
+		}
+
+		return () => {
+			if ( style && canSetStyles ) {
+				Object.keys( style )
+					.map( camelToKebab )
+					.forEach( styleName => {
+						editor.container.removeStyle( styleName );
+					} );
+			}
+		};
+	}, [ editor, status, style, type ] );
+
+	/**
+	 * Toggles read-only mode on runtime.
+	 */
+	useEffect( () => {
+		if ( editor && status === 'ready' && typeof readOnly === 'boolean' ) {
+			editor.setReadOnly( readOnly );
+		}
+	}, [ editor, status, readOnly ] );
+
+	/**
+	 * Registers listener to `instanceReady` event with high priority.
+	 * `initData` can be set only once during lifecycle of an editor instance.
+	 */
+	useEffect( () => {
+		if ( initData ) {
+			return registerEditorEventHandler( {
+				debug,
+				editor,
+				evtName: 'instanceReady',
+				handler: ( { editor } ) => {
+					editor.setData( initData );
```

Comment:
Maybe setting [`options.internal`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#method-setData) would solve the issues with undo (see overall comments about PR for more details) :thinking: 

```suggestion
					editor.setData( initData, {
						internal: true
					} );
```

Reply:
I've tried that but without luck...

The following `config` seems to be working fine though:

```js
{
	/**
	 * Prevents undo icon flickering.
	 */
	noSnapshot: true,

	/**
	 * Resets undo stack.
	 */
	callback: () => {
		editor.resetUndo();
	}
}
```